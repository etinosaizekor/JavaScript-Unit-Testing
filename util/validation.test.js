import { describe, it, expect } from "vitest";
import { validateNotEmpty } from "./validation";
import { ValidationError } from "./errors";

describe("validateNotError()", () => {
  it("should throw Error if empty string is provided", () => {
    const text = "";
    expect(() => {
      validateNotEmpty(text);
    }).toThrowError();
  });

  it("should throw Error if no text is provided", () => {
    expect(() => {
      validateNotEmpty();
    }).toThrowError();
  });

  it("should throw Error that was passed to it", () => {
    const text = "";
    const errMessage = 'Error occured'
    expect(() => {
      validateNotEmpty(text, errMessage);
    }).toThrowError(errMessage);
  });

  it("should throw Error if number is provided as parameter", () => {
    const text = 1;
    const errMessage = 'Error occured'
    expect(() => {
      validateNotEmpty(text, errMessage);
    }).toThrowError();
  });

  it.each(['hello', 'hello'+'world'])("should throw not Error valid text is provided as parameter to it", (text) => {
    // const text = "Hello";
    const errMessage = 'Error occured'
    expect(() => {
      validateNotEmpty(text, errMessage);
    }).not.toThrowError();
  });
});
