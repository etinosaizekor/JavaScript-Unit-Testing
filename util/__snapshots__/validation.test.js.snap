// Vitest Snapshot v1

exports[`validateNotError() > should throw Error that was passed to it > Error occured 1`] = `
ValidationError {
  "message": "Error occured",
}
`;
